platform/os_windows.go

```diff
@@ -189,6 +206,170 @@ func SetSdnRemoteArpMacAddress() error {
+		deviceid, err := ExecutePowershellCommand(cmd)
+
+		if err != nil {
+			return fmt.Errorf("error while executing powershell command to get device id of %s: %w", adapterName, err)
+		}
+		if deviceid == "" {
+			return fmt.Errorf("no network device found with %s in description", mellanoxSearchString)
+		}
+
+		log.Printf("Device ID found and Getting PNP device properites for %s", deviceid)
+		cmd = fmt.Sprintf("Get-PnpDeviceProperty -InstanceId \"%s\" | Where-Object KeyName -EQ \"DEVPKEY_Device_Driver\" | Select-Object -ExpandProperty Data", deviceid)
+		registryKeySuffix, err := ExecutePowershellCommand(cmd)
+		if err != nil {
+			return fmt.Errorf("error while executing powershell command to get registery suffix of device id %s: %w", deviceid, err)
+		}
+
+		registryKeyFullPath := registryKeyPrefix + registryKeySuffix
+		success, err := checkMellanoxPriorityVLANTagValueForV3(registryKeyFullPath, adapterName, regValueForVLANTag)
+		if err != nil {
+			log.Errorf("Error while checking registery value for PriorityVLANTag for adapater, continuing to reset the value: %v", err)
+		} else if success {
+			log.Printf("Mellanox PriorityVLANTag is already set to %v, skipping reset", regValueForVLANTag)
+			return nil
+		}
+
+		cmd = fmt.Sprintf("New-ItemProperty -Path \"%s\" -Name \"%s\" -Value %d -PropertyType String -Force", registryKeyFullPath, priorityVLANTagIdentifier, regValueForVLANTag)
+		_, err = ExecutePowershellCommand(cmd)
+		if err != nil {
+			return fmt.Errorf("error while executing powershell command to set Item property for device id  %s: %w", deviceid, err)
+		}
+
+		log.Printf("Restarting Mellanox network adapter for regkey change to take effect")
+		cmd = fmt.Sprintf("Restart-NetAdapter -Name \"%s\"", adapterName)
+		_, err = ExecutePowershellCommand(cmd)
+		if err != nil {
+			return fmt.Errorf("error while executing powershell command to restart net adapter  %s: %w", adapterName, err)
+		}
+		log.Printf("Successfully restarted Mellanox network adapter")
+
+		//double check if value is set or not
+		success, err = checkMellanoxPriorityVLANTagValueForV3(registryKeyFullPath, adapterName, regValueForVLANTag)
+		if err != nil {
+			return fmt.Errorf("error while executing powershell command to get item property  %s: %w", adapterName, err)
+		} else if !success {
+			return fmt.Errorf("for Mellanox CX-3 adapter, reg key was not found to be %d during post-check", regValueForVLANTag)
+		}
+		log.Printf("For Mellanox CX-3 adapters, the reg key set to %d", regValueForVLANTag)
+	}
+
+	return nil
+}
+
+// Checks if a Mellanox adapter's PriorityVLANTag value
+// for version 4 and up is set to the given expected value
+func checkMellanoxPriorityVLANTagValueForV4(adapterName string, expectedValue int) (bool, error) {
+	cmd := fmt.Sprintf(
+		"Get-NetAdapterAdvancedProperty | Where-Object { $_.RegistryKeyword -like \"%s\" -and $_.Name -eq \"%s\" } | Select-Object -ExpandProperty RegistryValue",
```

Comment:
use string literals to avoid escape characters
```suggestion
		`Get-NetAdapterAdvancedProperty | Where-Object { $_.RegistryKeyword -like "%s" -and $_.Name -eq "%s" } | Select-Object -ExpandProperty RegistryValue`,
```